pkg/utils/validation/admissionplugins/admissionplugins.go

```diff
@@ -145,8 +180,63 @@ func ValidateAdmissionPlugins(admissionPlugins []core.AdmissionPlugin, version s
 			if pointer.BoolDeref(plugin.Disabled, false) && admissionPluginsVersionRanges[plugin.Name].Required {
 				allErrs = append(allErrs, field.Forbidden(idxPath, fmt.Sprintf("admission plugin %q cannot be disabled", plugin.Name)))
 			}
+			if err := validateAdmissionPluginConfig(plugin, version, idxPath); err != nil {
+				allErrs = append(allErrs, err)
+			}
 		}
+
 	}
 
 	return allErrs
 }
+
+func validateAdmissionPluginConfig(plugin core.AdmissionPlugin, version string, fldPath *field.Path) *field.Error {
+	kubernetesVersion, err := semver.NewVersion(version)
+	if err != nil {
+		return field.Invalid(field.NewPath("spec", "kubernetes", "version"), version, err.Error())
+	}
+
+	switch plugin.Name {
+	case "PodSecurity":
+		if plugin.Config != nil {
+			config, err := runtime.Decode(codec, plugin.Config.Raw)
+			if err != nil {
+				if runtime.IsNotRegisteredError(err) {
+					return field.Invalid(fldPath.Child("config"), string(plugin.Config.Raw), "expected pod-security.admission.config.k8s.io/v1alpha1.PodSecurityConfiguration, pod-security.admission.config.k8s.io/v1beta1.PodSecurityConfiguration or pod-security.admission.config.k8s.io/v1.PodSecurityConfiguration")
+				}
+				return field.Invalid(fldPath.Child("config"), string(plugin.Config.Raw), fmt.Sprintf("cannot decode the given config: %s", err.Error()))
+			}
+
+			var errorString = "PodSecurityConfiguration apiVersion for Kubernetes version %q should be %q but got %q"
+
+			switch admissionConfigType := config.(type) {
+			case *admissionapiv1alpha1.PodSecurityConfiguration:
+				if !versionutils.ConstraintK8sEqual122.Check(kubernetesVersion) {
```

Comment:
Can't we use `>= 1.22` here? This would relax the necessity for people to upgrade bit?

Reply:
Good idea. Checked now and saw that v1alpha1 is not yet deprecated as of `v1.26.0`